solr/api/src/java/org/apache/solr/client/api/model/CollectionStatusResponse.java

```diff
@@ -82,21 +97,93 @@ public static class LeaderSummary {
 
     // Present with coreInfo=true || sizeInfo=true unless otherwise specified
     @JsonProperty public SegmentInfo segInfos;
+
+    private Map<String, Object> unknownFields = new HashMap<>();
+
+    @JsonAnyGetter
+    public Map<String, Object> unknownProperties() {
+      return unknownFields;
+    }
+
+    @JsonAnySetter
+    public void setUnknownProperty(String field, Object value) {
+      unknownFields.put(field, value);
+    }
   }
 
-  // Present with coreInfo=true || sizeInfo=true unless otherwise specified
+  // Present with segments=true || coreInfo=true || sizeInfo=true || fieldInfo=true unless otherwise
+  // specified
   public static class SegmentInfo {
-    // Present with coreInfo=true || sizeInfo=true unless otherwise specified
     @JsonProperty public SegmentSummary info;
 
+    // Present with segments=true || sizeInfo=true || fieldInfo=true
+    @JsonProperty public Map<String, SingleSegmentData> segments;
+
     // Present with rawSize=true
     @JsonProperty public RawSize rawSize;
 
-    // Present with fieldInfo=true....this seems pretty useless in isolation?  Is it maybe a bad
-    // param name?
+    // Present only with fieldInfo=true
     @JsonProperty public List<String> fieldInfoLegend;
   }
 
+  // Present with segment=true || sizeInfo=true
+  public static class SingleSegmentData {
+    @JsonProperty public String name;
+    @JsonProperty public Integer delCount;
+    @JsonProperty public Integer softDelCount;
+    @JsonProperty public Boolean hasFieldUpdates;
+    @JsonProperty public Long sizeInBytes;
+    @JsonProperty public Integer size;
+    // A date string of the form "2024-12-17T17:35:18.275Z"
+    @JsonProperty public String age;
```

Comment:
Can we use Instant?

Reply:
I've used `Date` for now based on PR discussion, see SOLR-17608.